### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by StarpTech**
Fix https://github.com/SamVerschueren/tsd-check/issues/6 Create service host and language server manually for fine tuning.

Before: `2617,4522ms`
After: `1018,9233ms`

```
Measure-Command { node .\dist\cli.js write-json-file-master }
```

### CODE REVIEW

**Comment by StarpTech** [comment]
break down to `~175,1671ms` 😈

**Comment by sindresorhus** [comment]
Can you elaborate on exactly what you're doing to improve performance? What kind of fine-tuning are you doing?

**Comment by StarpTech** [comment]
Hi @sindresorhus, there are two significant changes:

* Use `skipLibCheck` to skip all library `*.d.ts` files because we only want to test `.*test-d.ts` files
* Use only `getSemanticDiagnostics()` and `getSyntacticDiagnostics()` in order to check types. `getPreEmitDiagnostics()` will check [everyhting](https://github.com/Microsoft/TypeScript/blob/089b86af967ec0dcaa6d8d2ef6576e885605ecc5/src/compiler/program.ts#L202)

**Comment by StarpTech** [comment]
In my last commit, I removed a lot of boilerplate which weren't really necessary to archive the performance boost. The most important two tweaks are described above.

**Comment by StarpTech** [comment]
@sindresorhus could you retrigger the build for Node 8 it's hanged.

**Comment by sindresorhus** [comment]
What are we missing out on by not including `getGlobalDiagnostics`?

**Comment by sindresorhus** [comment]
> Use skipLibCheck to skip all library *.d.ts files because we only want to test .*test-d.ts files

What if the library we're testing depending on other library *.d.ts?

**Comment by StarpTech** [comment]
> What are we missing out on by not including getGlobalDiagnostics?

I didn't find any specification but it looks like diagnostics which doesn't belong to a file. Popular tools like https://github.com/TypeStrong/ts-node/blob/master/src/index.ts#L360 rely on the same configuration and it works great. Of course, it depends what we want to report but `getGlobalDiagnostics` isn't necessary for type hints.

> What if the library we're testing depending on other library *.d.ts?

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#new---skiplibcheck

This will still works but those definition files from other projects aren't included in the "type check" process because we can assume that the official definition files doesn't contain errors.

**Comment by StarpTech** [comment]
@sindresorhus @SamVerschueren can we merge it?

File: source/test/fixtures/failure-nested/index.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {expectType} from '../../..';
+import one from '.';
+import {} from './child.test-d';
```
[the remark this answers is not in the document]
Comment (StarpTech) [reply source/test/fixtures/failure-nested/index.test-d.ts]: We can but the I have to ignore a tslint rule

File: source/lib/compiler.ts [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+} from 'typescript';
 import {Diagnostic, Context} from './interfaces';
 
 // List of diagnostic codes that should be ignored
 const ignoredDiagnostics = new Set<number>([
 	1308 // Support top-level `await`
 ]);
 
-const loadConfig = () => {
+const loadConfig = () : CompilerOptions => {
```
[the remark this answers is not in the document]
Comment (StarpTech) [reply source/lib/compiler.ts]: Done

File: source/test/fixtures/failure-nested/index.test-d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (StarpTech) [reply source/test/fixtures/failure-nested/index.test-d.ts]: Done

**Comment by SamVerschueren** [comment]
Thank you!!! :tada:

**Pull request merged**